### PR TITLE
feat(storage, web): migrate web to js_interop to be compatible with WASM

### PR DESCRIPTION
**Pull request by Lyokone**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

#12027

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_storage/firebase_storage_web/lib/src/interop/storage_interop.dart [hunk 1]
```diff
@@ -8,225 +8,264 @@
...
+  external JSArray get items;
+  external JSString get nextPageToken;
+  external JSArray get prefixes;
 }
 
 // ignore: avoid_classes_with_only_static_members
 /// An enumeration of the possible string formats for upload.
 ///
-/// See: <https://firebase.google.com/docs/reference/js/firebase.storage#.StringFormat>
+/// See: <https://firebase.google.com/docs/reference/js/firebase.storage#.JSStringFormat>
```
Comment (russellwheatley) [inline packages/firebase_storage/firebase_storage_web/lib/src/interop/storage_interop.dart:248]:
```suggestion
/// See: <https://firebase.google.com/docs/reference/js/storage#stringformat>
```

File: packages/firebase_storage/firebase_storage_web/lib/src/interop/storage_interop.dart [hunk 2]
```diff
@@ -8,225 +8,264 @@
...
 
 @JS()
-external PromiseJsImpl<String> getDownloadURL(ReferenceJsImpl ref);
+@staticInterop
+external JSPromise getDownloadURL(ReferenceJsImpl ref);
 
 @JS()
-external PromiseJsImpl<String> getBlob(ReferenceJsImpl ref,
-    [int? maxDownloadSizeBytes]);
+@staticInterop
```
Comment (russellwheatley) [inline packages/firebase_storage/firebase_storage_web/lib/src/interop/storage_interop.dart:34]: Noticed we aren't putting the return type in a code comment anymore. Might be useful for our future selves?

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: packages/firebase_storage/firebase_storage_web/lib/src/interop/storage_interop.dart (hunk 2 above)
Comment (Lyokone) [reply packages/firebase_storage/firebase_storage_web/lib/src/interop/storage_interop.dart:34]: I've added typing in comments :)

File: packages/firebase_storage/firebase_storage_web/lib/src/interop/storage_interop.dart (hunk 1 above)
Comment (russellwheatley) [reply packages/firebase_storage/firebase_storage_web/lib/src/interop/storage_interop.dart]: @Lyokone  Just this one to merge and we should be good 😄

**Review by russellwheatley — APPROVED** [review]

**Pull request merged**